Functions/Quantification/xASL_quant_FSL.m

```diff
@@ -0,0 +1,573 @@
+% 7. Householding
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE: CBF_nocalib = xASL_quant_FSL(PWI, x);
+%
+% __________________________________
+% Copyright 2015-2023 ExploreASL 
+    
+
+    %% Admin
+    fprintf('%s\n','Quantification CBF using FSL BASIL/FABBER:');   
+
+	Tex_map = [];
+	ATT_map = [];
+	ABV_map = [];
+    
+    %% 1. Define paths
+    % For input, output, and options
+    pathFSLInput = fullfile(x.dir.SESSIONDIR, 'PWI4D_FSLInput.nii');
+    pathFSLOptions = fullfile(x.dir.SESSIONDIR, 'FSL_ModelOptions.txt');
+    dirFSLOutput = 'FSL_Output';
+	pathFSLOutput = fullfile(x.dir.SESSIONDIR, dirFSLOutput);
+	
+	% Define if BASIL or FABBER is used - multiTE needs FABBER 
+	if isfield(x.modules.asl, 'bMultiTE') && x.modules.asl.bMultiTE == 1
+		bUseFabber = 1;
+	else
+		bUseFabber = 0;
+	end
+        
+    %% 2. Delete previous output
+    xASL_adm_DeleteFileList(x.dir.SESSIONDIR, ['(?i)^' dirFSLOutput '.*$'], 1, [0 Inf]);
+    FolderList = xASL_adm_GetFileList(x.dir.SESSIONDIR, ['(?i)^' dirFSLOutput '.*$'], 'FPList', [0 Inf], 1);
+    for iFolder=1:numel(FolderList)
+        xASL_delete(FolderList{iFolder}, 1);
+    end
+    fprintf('%s\n', 'Note that any file not found warnings can be ignored, this pertains to the use of symbolic links by BASIL/FABBER');
+    
+    % Remove residual BASIL-related files
+    xASL_delete(pathFSLOptions);
+    xASL_delete(pathFSLInput);
+	xASL_delete(pathFSLOutput);
+    %xASL_adm_DeleteFileList(x.dir.SESSIONDIR, '(?i)^.*basil.*$', 1, [0 Inf]);
+    
+    %% 3. Write the PWI as Nifti file for Basil/Fabber to read as input
+    % FIXME would be good to have a brain mask at this point -> PM: if this would be a brainmask as well, we can skip creating a dummy input image here
+    PWI(isnan(PWI)) = 0;
+    
+    if ~x.modules.asl.bMultiPLD
+        % SinglePLD
+        xASL_io_SaveNifti(x.P.Path_PWI, pathFSLInput, PWI, [], 0); % use PWI path
+    else
+        % MultiPLD
+        xASL_io_SaveNifti(x.P.Path_PWI4D, pathFSLInput, PWI, [], 0); % use PWI4D path
+    end
+
+    %% 4. Create option_file that contains options which are passed to the FSL command
+    % Define basic paths
+
+	% Path to the options file
+	cmdlineOptions = [' --optfile ' xASL_adm_UnixPath(pathFSLOptions, ispc)];
+
+	% Path to input and output
+	if bUseFabber
+		cmdlineOptions = [cmdlineOptions ' --output=' xASL_adm_UnixPath(pathFSLOutput, ispc)];
+		cmdlineOptions = [cmdlineOptions ' --data=', xASL_adm_UnixPath(pathFSLInput, ispc)];
+	else
+		cmdlineOptions = [cmdlineOptions ' -o ' xASL_adm_UnixPath(pathFSLOutput, ispc)];
+		cmdlineOptions = [cmdlineOptions ' -i ' xASL_adm_UnixPath(pathFSLInput, ispc)];
+	end
+	
+	% basil_options is a character array containing CLI args for the Basil/Fabber command
+	BasilOptions = xASL_sub_FSLOptions(pathFSLOptions, x, bUseFabber, PWI, pathFSLInput, dirFSLOutput);
```

Comment:
This has more input options than allowed

Reply:
Good catch. Fixed.